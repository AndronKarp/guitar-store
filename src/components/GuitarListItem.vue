<template>
  <b-col>
    <router-link
      :to="{
        name: 'GuitarDescription',
        params: { slug: guitar.slug }
      }"
    >
      <b-card
        class="m-2"
        style="min-width: 280px;"
        :img-src="require(`../assets/${guitar.image}`)"
        :img-alt="guitar.model"
      >
        <b-card-title>{{ guitar.brand }} {{ guitar.model }}</b-card-title>
        <b-card-text>{{ guitar.price | currency }}</b-card-text>
      </b-card>
    </router-link>
  </b-col>
</template>

<script>
export default {
  props: {
    guitar: {
      type: Object,
      required: true
    }
  }
};
</script>

<style></style>
