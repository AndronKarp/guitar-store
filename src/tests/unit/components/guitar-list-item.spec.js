import { shallowMount, createLocalVue, RouterLinkStub } from "@vue/test-utils";
import GuitarListItem from "@/components/GuitarListItem";
import BootstrapVue from "bootstrap-vue";
import VueCurrencyFilter from "vue-currency-filter";

const localVue = createLocalVue();

localVue.use(BootstrapVue);
localVue.use(VueCurrencyFilter);

describe("GuitarListItem.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(GuitarListItem, {
      localVue,
      propsData: {
        guitar: {
          brand: "Gibson",
          model: "SG",
          image: "Gibson SG.png",
          price: 100,
          slug: "sg"
        }
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
  });

  test("renders guitar card", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("router link has correct path", () => {
    const routerLinkStub = wrapper.findComponent(RouterLinkStub);
    const path = routerLinkStub.props().to.name;
    expect(path).toBe("PageGuitarDescription");
  });

  test("router link passes slug param to the route", () => {
    const routerLinkStub = wrapper.findComponent(RouterLinkStub);
    const { slug } = routerLinkStub.props().to.params;
    expect(slug).toBe(wrapper.vm.guitar.slug);
  });
});
