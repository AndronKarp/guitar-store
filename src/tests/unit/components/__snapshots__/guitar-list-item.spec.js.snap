// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuitarListItem.vue renders guitar card 1`] = `
<b-col-stub tag="div"><a>
    <b-card-stub bodytag="div" titletag="h4" subtitletag="h6" subtitletextvariant="muted" headertag="div" footertag="div" imgsrc="" imgalt="SG" tag="div" class="m-2">
      <b-card-title-stub titletag="h4">Gibson SG</b-card-title-stub>
      <b-card-text-stub texttag="p"> 100</b-card-text-stub>
    </b-card-stub>
  </a></b-col-stub>
`;
